detox/test/e2e/06.device.test.js

```diff
@@ -39,6 +42,15 @@ describe('Device', () => {
     await expect(element(by.text('Hello!!!'))).toBeVisible();
   });
 
+  it('getViewHierarchyXml() - should return a valid UI hierarchy', async () => {
+    await device.launchApp({newInstance: true});
+    await element(by.text('Sanity')).tap();
+    await element(by.text('Say Hello')).tap();
+    const hierarchy = await device.getViewHierarchyXml();
+
+    jestExpect(hierarchy).toMatchSnapshot(`${device.getPlatform()} RN:${rnMinorVer}`);
```

Comment:
perfect
WDYT about also adding a step of XML verification (`isXML(hierarchy)` is truthy) using some XML parser like [fast-xml-parser](https://npmjs.com/package/fast-xml-parser)

Reply:
It is static. I've checked it before commiting the snapshot